muxrpc/handlers/tunnel/server/connect.go

```diff
@@ -21,6 +23,7 @@ type connectArg struct {
 type connectWithOriginArg struct {
 	connectArg
 	Origin refs.FeedRef `json:"origin"` // this should be clear from the shs session already
+	Caller refs.FeedRef `json:"caller"` // who started the call
```

Comment:
Is this the same as Portal? Given that there are just 3 peers, which one is the Caller? (Portal, Target, or Origin?)

Reply:
This addition was a bit inconsiderate and out of frustration. Just to recap my understanding:

* Portal: the room server.
* Target: the peer the initiator (or origin) want's to connect to.
* Origin: the peer who made the initial `tunnel.connect` towards the room.

What I have been seeing is that portal == origin and the missing bit for the receiver of the request (the target) was who the caller was, which can only be learned after shs handshake completes, unless the server already notifies the target via the proposed new `caller` argument.

I just added a new one because I wasn't sure if the JS implementation needs any of these so I didnt want to mess with this but from your comment I assume that it's okay to fix Origin to be Caller and not Portal, then.

My frustration stemmed from this:

* Portal: is irrelevant for the room as well as the target
  * the room knows who it is
  * the target/receiver knows who the request is from via shs auth
* Target: only relevant for the room and doesn't need to be forwarded (the target knows who it is)
* Origin: only info that needs to be forwarded to the target but should be set by the origin itself (can be set by the server which takes it from shs auth)

Reply:
I think the redundancy you felt with `Portal` was because ssb-tunnel was originally built so that one `index.js` can be used for all of those 3 roles.